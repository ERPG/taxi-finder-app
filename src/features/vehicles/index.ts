import VehiclesFeature from "./vehicles";

export default VehiclesFeature;