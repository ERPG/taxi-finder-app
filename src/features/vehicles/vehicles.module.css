.container {
    display: flex;
    flex-direction: row;
    gap: 20px;
    height: 100vh;
    padding: 20px;
    box-sizing: border-box;
}

.mapContainer {
    flex: 1;
    height: 392px;
    width: 569px;
    padding: 10px;
}

.tableContainer {
    flex: 1;
    height: auto;
}

@media (max-width: 1200px) {
    .mapContainer {
        width: 320px;
        height: 392px;
    }
}

@media (max-width: 993px) {
    .container {
        flex-direction: column;
    }

    .mapContainer {
        width: 100%;
    }
}

@media (max-width: 768px) {
    .mapContainer {
        width: 100%;
        height: 60vh;
    }
}