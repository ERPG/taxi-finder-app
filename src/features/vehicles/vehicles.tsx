import { Suspense, useState } from "react";
import useErrorBoundary from "../../shared/components/error-boundary/error-boundary";
import VehiclesTable from "./components/vehicles-table/vehicles-table";
import VehiclesMap from "./components/vehicles-map/vehicles-map";
import { useVehicles } from "./hooks/use-vehicles";

import styles from "./vehicles.module.css";
import { PAGE_LIMIT } from "../../shared/utils/constants";
import ErrorFallback from "@/shared/components/error-fallback/error-fallback";
import { VehicleSelectionProvider } from "./contexts/select-vehicle.context";

const VehiclesFeature = () => {
    const [currentPage, setCurrentPage] = useState<number>(1);
    const { ErrorBoundaryWrapper } = useErrorBoundary();

    const { data: vehicles, isLoading, isError, refetch } = useVehicles(PAGE_LIMIT, currentPage);

    if (isLoading) return <p>Loading...</p>;
    if (isError) return <ErrorFallback refetch={refetch} />;

    const handlePageChange = (page: React.SetStateAction<number>) => {
        setCurrentPage(page);
    };

    return (
        <ErrorBoundaryWrapper>
            <Suspense fallback={<div>Loading vehicles...</div>}>
                <VehicleSelectionProvider>
                    <div className={styles.container}>
                        <div className={styles.mapContainer}>
                            <VehiclesMap vehicles={vehicles?.data} />
                        </div>
                        <div className={styles.tableContainer}>
                            <VehiclesTable
                                vehicles={vehicles}
                                currentPage={currentPage}
                                onCurrentPageChange={handlePageChange}
                                isLoading={isLoading}
                            />
                        </div>
                    </div>
                </VehicleSelectionProvider>
            </Suspense>
        </ErrorBoundaryWrapper>
    );
};

export default VehiclesFeature;
