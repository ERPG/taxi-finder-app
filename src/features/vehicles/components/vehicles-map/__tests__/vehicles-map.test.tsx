import { describe, expect, vi } from "vitest";
import { render } from "@testing-library/react";
import VehiclesMap from "../vehicles-map";
import { mockMapVehicles } from "../../../../../tests/__mocks__/vehicles";

describe('vehicles map component suite', () => {

    const props = {
        vehicles: mockMapVehicles,
    };

    afterEach(() => {
        vi.clearAllMocks();
    });

    test("Should match snapshot", () => {
        const { container } = render(<VehiclesMap {...props} />);
        expect(container).toMatchSnapshot();
    });

    test("should render the map with correct center and zoom", () => {
        render(<VehiclesMap {...props} />);

        expect(document.querySelector(".leaflet-container")).toBeInTheDocument();
        expect(document.querySelector(".leaflet-tile")).toBeInTheDocument();
    });

    test("should render markers for all vehicles", () => {
        render(<VehiclesMap {...props}/>);

        const markers = document.querySelectorAll(".leaflet-marker-icon");
        expect(markers).toHaveLength(props.vehicles.length);
    });
});

