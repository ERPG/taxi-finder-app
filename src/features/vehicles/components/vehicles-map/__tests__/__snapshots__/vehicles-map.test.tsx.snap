// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`vehicles map component suite > Should match snapshot 1`] = `
<div>
  <div
    class="leaflet-container leaflet-touch leaflet-grab leaflet-touch-drag leaflet-touch-zoom"
    style="height: 100%; width: 100%; min-height: 392px; position: relative;"
    tabindex="0"
  >
    <div
      class="leaflet-pane leaflet-map-pane"
      style="left: 0px; top: 0px;"
    >
      <div
        class="leaflet-pane leaflet-tile-pane"
      >
        <div
          class="leaflet-layer "
          style="z-index: 1;"
        >
          <div
            class="leaflet-tile-container leaflet-zoom-animated"
            style="z-index: 18; left: 0px; top: 0px;"
          >
            <img
              alt=""
              class="leaflet-tile"
              src="https://a.tile.openstreetmap.org/18/138351/84723.png"
              style="width: 256px; height: 256px; left: -124px; top: -115px;"
            />
          </div>
        </div>
      </div>
      <div
        class="leaflet-pane leaflet-overlay-pane"
      />
      <div
        class="leaflet-pane leaflet-shadow-pane"
      />
      <div
        class="leaflet-pane leaflet-marker-pane"
      >
        <div
          class="leaflet-marker-icon custom-icon leaflet-zoom-hide leaflet-interactive"
          role="button"
          style="margin-left: -20px; margin-top: -20px; width: 40px; height: 40px; left: -587px; top: -173px; z-index: -173;"
          tabindex="0"
        >
          <div
            style="width:40px;height:40px;background-color:#0066cc;border-radius:50%;display:flex;align-items:center;justify-content:center;box-shadow:0 2px 5px rgba(0,0,0,0.3)"
          >
            <svg
              color="white"
              fill="none"
              height="20"
              viewBox="0 0 24 24"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14.842 3a3 3 0 012.814 1.96l1.698 4.605.21.435H22v2h-1.49c.323.817.49 1.69.49 2.572V21h-3v-2H6v2H3v-6.724A7 7 0 013.38 12H2v-2h2.242l.163-.362 1.837-4.725A3 3 0 019.038 3h5.804zm3.469 9H5.539l-.106.241a5 5 0 00-.424 1.735l-.009.3V17h14v-2.428c0-.754-.17-1.498-.499-2.177l-.19-.395zM7.5 13a1.5 1.5 0 110 3 1.5 1.5 0 010-3zm9 0a1.5 1.5 0 110 3 1.5 1.5 0 010-3zm-1.658-8H9.038a1 1 0 00-.932.638L6.408 10h10.979L15.78 5.653a1 1 0 00-.813-.645L14.842 5z"
                fill="currentColor"
              />
            </svg>
          </div>
        </div>
        <div
          class="leaflet-marker-icon custom-icon leaflet-zoom-hide leaflet-interactive"
          role="button"
          style="margin-left: -20px; margin-top: -20px; width: 40px; height: 40px; left: 587px; top: 172px; z-index: 172;"
          tabindex="0"
        >
          <div
            style="width:40px;height:40px;background-color:#0066cc;border-radius:50%;display:flex;align-items:center;justify-content:center;box-shadow:0 2px 5px rgba(0,0,0,0.3)"
          >
            <svg
              color="white"
              fill="none"
              height="20"
              viewBox="0 0 24 24"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14.842 3a3 3 0 012.814 1.96l1.698 4.605.21.435H22v2h-1.49c.323.817.49 1.69.49 2.572V21h-3v-2H6v2H3v-6.724A7 7 0 013.38 12H2v-2h2.242l.163-.362 1.837-4.725A3 3 0 019.038 3h5.804zm3.469 9H5.539l-.106.241a5 5 0 00-.424 1.735l-.009.3V17h14v-2.428c0-.754-.17-1.498-.499-2.177l-.19-.395zM7.5 13a1.5 1.5 0 110 3 1.5 1.5 0 010-3zm9 0a1.5 1.5 0 110 3 1.5 1.5 0 010-3zm-1.658-8H9.038a1 1 0 00-.932.638L6.408 10h10.979L15.78 5.653a1 1 0 00-.813-.645L14.842 5z"
                fill="currentColor"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="leaflet-pane leaflet-tooltip-pane"
      />
      <div
        class="leaflet-pane leaflet-popup-pane"
      />
    </div>
    <div
      class="leaflet-control-container"
    >
      <div
        class="leaflet-top leaflet-left"
      >
        <div
          class="leaflet-control-zoom leaflet-bar leaflet-control"
        >
          <a
            aria-disabled="true"
            aria-label="Zoom in"
            class="leaflet-control-zoom-in leaflet-disabled"
            href="#"
            role="button"
            title="Zoom in"
          >
            <span
              aria-hidden="true"
            >
              +
            </span>
          </a>
          <a
            aria-disabled="false"
            aria-label="Zoom out"
            class="leaflet-control-zoom-out"
            href="#"
            role="button"
            title="Zoom out"
          >
            <span
              aria-hidden="true"
            >
              −
            </span>
          </a>
        </div>
      </div>
      <div
        class="leaflet-top leaflet-right"
      />
      <div
        class="leaflet-bottom leaflet-left"
      />
      <div
        class="leaflet-bottom leaflet-right"
      >
        <div
          class="leaflet-control-attribution leaflet-control"
        >
          <a
            href="https://leafletjs.com"
            title="A JavaScript library for interactive maps"
          >
            Leaflet
          </a>
           
          <span
            aria-hidden="true"
          >
            |
          </span>
           © OpenStreetMap contributors
        </div>
      </div>
    </div>
    <div />
    <div />
  </div>
</div>
`;
